source/common/upstream/cds_api_impl.cc

```diff
@@ -1,65 +1,53 @@
 #include "common/upstream/cds_api_impl.h"
 
-#include <chrono>
 #include <string>
-#include <vector>
 
-#include "common/common/assert.h"
-#include "common/config/cds_json.h"
+#include "common/config/subscription_factory.h"
```

Comment:
Yeah. The rule-of-thumb we're trying to do I think is IWYU (although admittedly, this is rarely complete as we don't have automated enforcement). BUILD deps should largely reflect the headers in the `.{h,cc}`.